### DISCUSSION TITLE
'Nodemon' is not recognized as an internal or external command

### DISCUSSION THREAD

**Discussion by elitaylor** [community/community#131636] [Programming Help]
### Body

I am trying to build a web scraper using JavaScript, Node.js, Express, Axios, and Cheerio. Till now I have configured package.json file and it's look like this:
```
{
  "name": "whatsapp-scraper",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "start": "nodemon index.js"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "axios": "^1.6.7",
    "cheerio": "^1.0.0-rc.12",
    "express": "^4.17.1"
  }
}
```
After that I have configured the index.js file and in the file, I put one of a test [WhatsApp group](https://whatagrouplinks.com/) website to scrap the HTML of the site and the given WhatsApp chat links. The index.js file looks like this:

```
const PORT = 8000
const axios = require('axios')
const cheerio = require('cheerio')
const express = require('express')
const {response} = require("express");

const app = express()

const url = 'https://whatagrouplinks.com/'

axios(url)
.then(response => {
    const html = response.data
    console.log(html)
})

app.listen(PORT, () => console.log(`server running on PORT ${PORT}`))
```
Now after configuring the index.js file, it should return the query with the results in the form of HTML of the page and also the links. But in the terminal it's returning this code:

```
'nodemon' is not recognized as an internal or external command,
operable program or batch file.
```
Does anyone know how to fix this.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ebndev** [comment]
Hey @elitaylor, thanks for being a part of the GitHub Community!

nodemon is an npm package so you need to install it via the terminal or add it to your dependencies ie. `"nodemon": "^1.19.1` (whatever the latest version is.

Let us know if that works or if you have other questions 😄

**Comment by DharshiBalasubramaniyam** [comment]
@elitaylor Its seems like you have not installed the nodemon in your project. 
"dependencies": {
    "axios": "^1.6.7",
    "cheerio": "^1.0.0-rc.12",
    "express": "^4.17.1"
  }
nodemon is missing in the dependencies.

So you need to install nodemon.

Open your terminal in the root directory of your project and run the following command.

`npm install nodemon`

This command will install the nodemon. After installing make sure that nodemon is listed under dependencies in package.json file.

After installing nodemon, you should be able to run your application without encountering the error.

**Comment by ebndev** [comment]
Hi @DharshiBalasubramaniyam, thanks for taking the time to answer!

@elitaylor if the response above helped, please feel encouraged to mark it as the answer. If not, let us know if you have further questions or post the solution if you've resolved it on your own. It would be helpful for other users who might experience this in the future 😄
